contracts/pcv/compound/ERC4626CErc20Wrapper.sol

```diff
@@ -0,0 +1,240 @@
+        _burn(owner, shares);
+
+        // redeem cTokens
+        uint256 balanceBefore = cTokenUnderlying.balanceOf(address(this));
+        require(cToken.redeem(shares) == 0, "ERC4626CErc20Wrapper: error on cToken.redeemUnderlying");
+        assets = cTokenUnderlying.balanceOf(address(this)) - balanceBefore;
+
+        // transfer underlying to receiver
+        cTokenUnderlying.transfer(receiver, assets);
+
+        // emit event
+        emit Withdraw(msg.sender, receiver, assets, shares);
+
+        return assets;
+    }
+
+    /*///////////////////////////////////////////////////////////////
+                ERC4626 Vault Accounting Logic
+    //////////////////////////////////////////////////////////////*/
+
+    /// @notice The amount of shares that the vault would
+    /// exchange for the amount of assets provided, in an
+    /// ideal scenario where all the conditions are met.
+    function convertToShares(uint256 assets) public view returns(uint256 shares) {
+        return assets * 1e18 / cToken.exchangeRateStored();
+    }
+
+    /// @notice The amount of assets that the vault would
+    /// exchange for the amount of shares provided, in an
+    /// ideal scenario where all the conditions are met.
+    function convertToAssets(uint256 shares) public view returns(uint256 assets) {
+        return shares * cToken.exchangeRateStored() / 1e18;
+    }
+
+    /// @notice Total number of underlying assets that can
+    /// be deposited by `owner` into the Vault, where `owner`
+    /// corresponds to the input parameter `receiver` of a
+    /// `deposit` call.
+    function maxDeposit(address/* owner*/) external pure returns(uint256 maxAssets) {
+        return type(uint256).max;
+    }
+
+    /// @notice Allows an on-chain or off-chain user to simulate
+    /// the effects of their deposit at the current block, given
+    /// current on-chain conditions.
+    function previewDeposit(uint256 assets) external view returns(uint256 shares) {
+        return convertToShares(assets);
+    }
+
+    /// @notice Total number of underlying shares that can be minted
+    /// for `owner`, where `owner` corresponds to the input
+    /// parameter `receiver` of a `mint` call.
+    function maxMint(address/* owner*/) external pure returns(uint256 maxShares) {
+        return type(uint256).max;
+    }
+
+    /// @notice Allows an on-chain or off-chain user to simulate
+    /// the effects of their mint at the current block, given
+    /// current on-chain conditions.
+    function previewMint(uint256 shares) external view returns(uint256 assets) {
+        return convertToAssets(shares);
+    }
+
+    /// @notice Total number of underlying assets that can be
+    /// withdrawn from the Vault by `owner`, where `owner`
+    /// corresponds to the input parameter of a `withdraw` call.
+    function maxWithdraw(address owner) external view returns(uint256 maxAssets) {
+        uint256 sharesOwned = balanceOf(owner);
```

Comment:
This won’t be exact due to rounding

Reply:
I'll add comments where amounts might now be totally exact